lib/instrumentation/modules/aws-sdk/sns.js

```diff
@@ -0,0 +1,102 @@
+const constants = require('../../../constants')
+
+const TYPE = 'messaging'
+const SUBTYPE = 'sns'
+const ACTION = 'PUBLISH'
+const ACCESS_POINT = 'accesspoint'
+const PHONE_NUMBER = '<PHONE_NUMBER>'
+
+function getRegionFromRequest (request) {
+  return request && request.service &&
+        request.service.config && request.service.config.region
+}
+
+function getSpanNameFromRequest (request) {
+  const topicName = getDestinationNameFromRequest(request)
+  return `SNS ${ACTION} ${topicName}`
+}
+
+function getMessageContextFromRequest (request) {
+  return {
+    queue_name: getDestinationNameFromRequest(request)
+  }
+}
+
+function getMessageDestinationContextFromRequest (request) {
+  return {
+    resource: `${SUBTYPE}/${getDestinationNameFromRequest(request)}`,
+    type: TYPE,
+    name: SUBTYPE,
```

Comment:
"destination": {
        "resource": "sns/trentm-play-topic1",
        "type": "messaging",
        "name": "sns",
        "cloud": {
            "region": "us-west-1"
        }
    }

This should be formatted as:

    "destination": {
        "service": {
            "resource": "sns/trentm-play-topic1",
            "type": "messaging",
            "name": "sns"
        },
        "cloud": {
            "region": "us-west-1"
        }
    },

Reply:
- Also, `context.destination.service.{address,port}` can be extracted from
  `request.service.endpoint.{hostname,port}` or `request.httpRequest.endpoint.`
  (spec: https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-messaging.md#span-destination-fields)

Reply:
Adjusted destination context structure and added address and port fields.  (See `AWS SNS: End to End Test -> API: publish` test for full assertions on generated spans.